overlord/state/taskrunner.go

```diff
@@ -150,12 +166,15 @@ func (r *TaskRunner) Ensure() {
 			// a task can be in RunningStatus even when it
 			// is not started yet (like when the daemon
 			// process restarts)
-			if _, ok := r.tombs[t.ID()]; ok {
+			if tomb, ok := r.tombs[t.ID()]; ok {
+				if isPointless(t) {
```

Comment:
Also don't understand this. The fact something it depends on is in error means nothing for this specific task. The stated dependency is the other way around.